test/WebJobs.Script.Tests/Middleware/HttpThrottleMiddlewareTests.cs

```diff
@@ -0,0 +1,302 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.Azure.WebJobs.Script.Description;
+using Microsoft.Azure.WebJobs.Script.Diagnostics;
+using Microsoft.Azure.WebJobs.Script.Scale;
+using Microsoft.Azure.WebJobs.Script.WebHost;
+using Microsoft.Azure.WebJobs.Script.WebHost.Middleware;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Extensions.Primitives;
+using Microsoft.WebJobs.Script.Tests;
+using Moq;
+using Xunit;
+
+namespace Microsoft.Azure.WebJobs.Script.Tests
+{
+    public class HttpThrottleMiddlewareTests
+    {
+        private readonly HttpThrottleMiddleware _middleware;
+        private readonly Mock<IMetricsLogger> _metricsLogger;
+        private readonly Mock<HostPerformanceManager> _performanceManager;
+        private readonly FunctionDescriptor _functionDescriptor;
+        private readonly TestLoggerProvider _loggerProvider;
+        private readonly LoggerFactory _loggerFactory;
+        private readonly Mock<IScriptJobHost> _scriptHost;
+        private int _throttleMetricCount;
+        private HttpRequestQueue _requestQueue;
+        private HttpOptions _httpOptions;
+
+        public HttpThrottleMiddlewareTests()
```

Comment:
Ported all the tests over from https://github.com/Azure/azure-webjobs-sdk-extensions/pull/463.